contents/docs/feature-flags/evaluation-tags.mdx

```diff
@@ -0,0 +1,253 @@
+
+In your PostHog organization settings, certain tags can be designated as "evaluation tags". These are regular tags that have been given the special ability to act as environment constraints.
+
+### Step 2: Apply evaluation tags to flags
+
+When creating or editing a feature flag:
+
+1. Navigate to the tags section
+2. Click the bolt icon (⚡) to configure evaluation constraints
+3. Select which tags should act as evaluation constraints
+4. Selected evaluation tags will display with a green background and bolt icon
+
+> **Remember:** Setting tags in the PostHog app is only half the setup. Your application needs to know which environment it's in, which requires updating your SDK configuration (Step 3).
+
+### Step 3: Configure your SDKs (Critical!)
+
+**This step is essential** - After setting evaluation tags in the PostHog app, you must update your SDK configuration to tell your application which environment it's running in. Without this, your application won't know its deployment context and evaluation tags won't work.
+
+Update your SDK initialization to include evaluation environments:
+
+#### JavaScript/TypeScript
+```javascript
+posthog.init('YOUR_API_KEY', {
+    api_host: 'https://app.posthog.com',
+    evaluation_environments: ['production', 'web', 'checkout']
+})
+```
+
+#### Python
+```python
+posthog = Posthog(
+    'YOUR_API_KEY',
+    host='https://app.posthog.com',
+    evaluation_environments=['production', 'api', 'backend']
+)
+```
+
+#### React
+```jsx
+<PostHogProvider 
+    apiKey='YOUR_API_KEY'
+    options={{
+        api_host: 'https://app.posthog.com',
+        evaluation_environments: ['production', 'web', 'app']
+    }}
+>
+    <App />
+</PostHogProvider>
+```
+
+#### Node.js
+```javascript
+const posthog = new PostHog('YOUR_API_KEY', {
+    host: 'https://app.posthog.com',
+    evaluation_environments: ['staging', 'api', 'backend']
+})
+```
+
+## How evaluation works
+
+When a flag evaluation request is made:
+
+1. **SDK sends environment tags**: The SDK includes its configured `evaluation_environments` in the request
+2. **PostHog filters flags**: Only flags matching these criteria are evaluated:
+   - Flags with no evaluation tags (backward compatibility)
+   - Flags with empty evaluation tags
+   - Flags where at least one evaluation tag matches the provided environments
+3. **Results returned**: Only the filtered flags are returned to the SDK
+
+### Example scenario
+
+Consider these feature flags:
+- Flag A: No evaluation tags → Evaluates for all requests
+- Flag B: Evaluation tags `["production", "web"]` → Only evaluates when SDK provides "production" OR "web"
+- Flag C: Evaluation tags `["staging"]` → Only evaluates when SDK provides "staging"
+
+If an SDK is configured with `evaluation_environments: ["production", "mobile"]`:
+- ✅ Flag A evaluates (no constraints)
+- ✅ Flag B evaluates ("production" matches)
+- ❌ Flag C does NOT evaluate (no matching tags)
+
+## Best practices
+
+### 1. Start simple
```

Comment:
I think we can lose the numbers here. We have a lot of numbered headings already for steps to follow, this might be distracting.